generate_index_html.py

```diff
@@ -0,0 +1,118 @@
+import requests
+import re
+import subprocess
+import argparse
+from packaging.requirements import Requirement
+from packaging.version import Version
+
+def get_custom_released_package_versions(bucket: str, folder: str) -> dict:
+    data = subprocess.check_output([f"gsutil ls -lh 'gs://{bucket}/{folder}/' | grep -E '\.whl|\.tar\.gz'"], shell=True)
+    # Currently similar strings are correctly parsed:
+    # - apache_airflow_providers_cncf_kubernetes-1.0.2rc1-py3-none-any.whl
+    # - apache_airflow_providers_cncf_kubernetes-1.0.2+astro+1-py3-none-any.whl
+    # - apache_airflow_providers_cncf_kubernetes-1.0.2-py3-none-any.tar.gz
+    # - apache_airflow_providers_cncf_kubernetes-1.0.2-py3-none-any.whl
+    # Note - platform information is not handled atm.
+    pattern = re.compile("([\w\W]+?)  ([\w\W]+?)  gs://([\w\W]+?(?=/))/([\w\W]+?(?=/))/(([\w\W]+?)-(\d+\.\d+\.\d+(rc\d+)?(\+astro\.\d+)?)-py3-none-any(?:\.whl|\.tar\.gz))")
+    matches = pattern.findall(data.decode())
+    version_to_files = {}
```

Comment:
Comment explaining what could be in version_to_files would be appreciated. e.g what's match[6], match[4] and match[1]

Reply:
I have added what each match group will hold for a given example string. PTAL.